src/librustdoc/clean/types.rs

```diff
@@ -1534,10 +1534,10 @@ impl Type {
         matches!(self, Type::Path { path: Path { res: Res::Def(DefKind::TyAlias, _), .. } })
     }
 
-    /// Check if two types are "the same" for documentation purposes.
+    /// Check if this type is a subtype of another type for documentation purposes.
     ///
     /// This is different from `Eq`, because it knows that things like
-    /// `Placeholder` are possible matches for everything.
+    /// `Placeholder` and generics have special subtyping rules.
```

Comment:
```suggestion
    /// `Infer` and generics have special subtyping rules.
```

I'm assuming this enum variant got renamed at some point.

Reply:
I did small research about this, and found this place, do you think it also will be reasonable to replace `Placeholder` with `Infer` here?

https://github.com/rust-lang/rust/blob/e1b9081e699065badfc1a9419ec9566e5c8615c4/src/librustdoc/clean/types.rs#L1589-L1590

Reply:
that would make the comment less helpful IMO, as the name of the type is clear from the following line anyways.  It's also using "Placeholders" as just a regular word, since there's no inline code block, unlike in the doc comment.  The doc comment should make sense without the context of the code, this only needs to make sense in context.